lib/ansible/executor/task_executor.py

```diff
@@ -655,8 +655,12 @@ def _evaluate_failed_when_result(result):
 
             if 'ansible_facts' in result and self._task.action not in C._ACTION_DEBUG:
                 if self._task.action in C._ACTION_WITH_CLEAN_FACTS:
-                    vars_copy.update(result['ansible_facts'])
+                    if self._task.delegate_to and self._task.delegate_facts:
+                        vars_copy.update({'_ansible_delegated_vars': result['ansible_facts']})
```

Comment:
This will replace the whole `_ansible_delegated_vars` key. Not sure if that was your intention or not.

I think you might want `vars_copy['_ansible_delegated_vars'].update(result['ansible_facts'])`